lib/HttpClient.ts

```diff
@@ -88,7 +88,10 @@ export class HttpClient {
     private _httpProxyBypassHosts: RegExp[];
     private _allowRedirects: boolean = true;
     private _maxRedirects: number = 50
-
+    private _agent;
```

Comment:
kill line 11 hard coded max.  instead, just default your options to 100 if not set (or should we not set)?  never liked the hard code setting which we never removed.  since you're added maxConnections options, let's take it out.

Reply:
default of 100 seems reasonable to me, will update PR.